e2e/tests/test_e2e.py

```diff
@@ -891,6 +891,102 @@ def test_zzz_taint_based_eviction(self):
         new_master_node = nm[0]
         self.assert_distributed_pods(new_master_node, new_replica_nodes, cluster_label)
 
+    @timeout_decorator.timeout(TEST_TIMEOUT_SEC)
+    def test_node_affinity(self):
+        '''
+           Add label to a node and update postgres cluster spec to deploy only on a node with that label
+        '''
+        k8s = self.k8s
+        cluster_label = 'application=spilo,cluster-name=acid-minimal-cluster'
+
+        # get nodes of master and replica(s)
+        current_master_node, current_replica_nodes = k8s.get_pg_nodes(cluster_label)
+
+        # label node with environment=postgres
+        node_label_body = {
+            "metadata": {
+                "labels": {
+                    "node-affinity-test": "postgres"
+                }
+            }
+        }
+
+        try:
+            # patch current master node with the label
+            print('patching master node: {}'.format(current_master_node))
+            k8s.api.core_v1.patch_node(current_master_node, node_label_body)
+
+            # add node affinity to cluster
+            patch_node_affinity_config = {
+                "spec": {
+                    "nodeAffinity" : {
+                        "requiredDuringSchedulingIgnoredDuringExecution": {
+                            "nodeSelectorTerms": [
+                                {
+                                    "matchExpressions": [
+                                        {
+                                            "key": "node-affinity-test",
+                                            "operator": "In",
+                                            "values": [
+                                                "postgres"
+                                            ]
+                                        }
+                                    ]
+                                }
+                            ]
+                        }
+                    }
+                }
+            }
+
+            k8s.api.custom_objects_api.patch_namespaced_custom_object(
+                group="acid.zalan.do",
+                version="v1",
+                namespace="default",
+                plural="postgresqls",
+                name="acid-minimal-cluster",
+                body=patch_node_affinity_config)
+
+            # bounce replica and check if it has migrated to proper node that has a label
+            k8s.api.core_v1.delete_namespaced_pod("acid-minimal-cluster-1", "default")
+
+            # wait for operator to apply node affinity changes
+            time.sleep(120)
```

Comment:
doesn't the previous pod trigger a failover? maybe you can use the `assert_failover` moethod then. Instead of using `sleep` we have recently switched to using `eventuallyEqual` and `eventuallyTrue` that retry and potentially finish way sooner than just waiting.

Reply:
Fixed. Removed hard coded sleep. and removed second sleep, that was not necessary there.